src/can.c

```diff
@@ -186,6 +186,12 @@ static void thingset_can_report_rx_cb(const struct device *dev, struct can_frame
                 (struct thingset_can_rx_context *)buffer->user_data;
             if (context->seq++ == frame->data[0]) {
                 int size = can_dlc_to_bytes(frame->dlc) - 1;
+                if (buffer->len + size > buffer->size) {
+                    LOG_WRN("Discarding packetised message from %x for data ID %x as it is too large.",
```

Comment:
Can you add `0x` to make clear those are hex values?

```suggestion
                    LOG_WRN("Discarding packetised message from 0x%X for data ID 0x%X as it is too large.",
```